frontend/src/components/Forms/PartyRelationships/AddPartyRelationshipForm.js

```diff
@@ -23,13 +28,58 @@ const defaultProps = {
   mine: {},
 };
 
-const validate = (values) => {
+const checkDatesForOverlap = (values, props) => {
+  const existingAppointments = props.partyRelationships.filter(
+    ({ mine_party_appt_type_code }) =>
+      mine_party_appt_type_code === props.partyRelationshipType.mine_party_appt_type_code
+  );
+  const newAppt = { start_date: null, end_date: null, ...values };
+
+  // Sort all appointments into ascending order by start_date
+  const toDate = (dateString) => moment(dateString, "YYYY-MM-DD").toDate();
+  const allAppointments = [...existingAppointments, newAppt].sort((a, b) =>
+    toDate(a.start_date) > toDate(b.start_date) ? 1 : -1
+  );
+
+  const errorMessages = {};
+  for (let i = 0; i < allAppointments.length - 1; i += 1) {
+    const current = allAppointments[i];
+    const next = allAppointments[i + 1];
+
```

Comment:
If bad data is somehow imported into the system (two mine managers overlap, but not the current one being added) , won't this fail?

Reply:
This constraint exists at the database level at the moment. We can't simultaneously enforce "good" data while ignoring "bad" data without severely making a mess of our validations